spec/01-unit/08-entities_schemas_spec.lua

```diff
@@ -449,6 +452,230 @@ describe("Entities Schemas", function()
+        slots = 10,
+        orderlist = { 9,7,5,3,1,2,4,6,8 }   -- too short (9)
+      }
+      valid, errors, check = validate_entity(data, upstreams_schema)
+      assert.False(valid)
+      assert.Nil(errors)
+      assert.are.equal("size mismatch between 'slots' and 'orderlist'",check.message)
+
+      data = { 
+        name = "valid.host.name",
+        slots = 10,
+        orderlist = { 9,7,5,3,1,2,4,6,8,10,11 }   -- too long (11)
+      }
+      valid, errors, check = validate_entity(data, upstreams_schema)
+      assert.False(valid)
+      assert.Nil(errors)
+      assert.are.equal("size mismatch between 'slots' and 'orderlist'",check.message)
+
+      data = { 
+        name = "valid.host.name",
+        slots = 10,
+        orderlist = { 9,7,5,3,1,2,4,6,8,8 }   -- a double value (2x 8, no 10)
+      }
+      valid, errors, check = validate_entity(data, upstreams_schema)
+      assert.False(valid)
+      assert.Nil(errors)
+      assert.are.equal("invalid orderlist",check.message)
+
+      data = { 
+        name = "valid.host.name",
+        slots = 10,
+        orderlist = { 9,7,5,3,1,2,4,6,8,11 }   -- a hole (10 missing)
+      }
+      valid, errors, check = validate_entity(data, upstreams_schema)
+      assert.False(valid)
+      assert.Nil(errors)
+      assert.are.equal("invalid orderlist",check.message)
+    end)
+
+  end)
+  
+  --
+  -- TARGETS
+  --
+
+  describe("Targets", function()
+    local weight_default, weight_min, weight_max = 100, 0, 1000
+    local default_port = 8000
+    
+    it("should validate the required 'target' field", function()
+      local valid, errors, check
+
+      valid, errors, check = validate_entity({}, targets_schema)
+      assert.False(valid)
+      assert.equal(errors.target, "target is required")
+      assert.is_nil(check)
+
+      local names = { "valid.name", "valid.name:8080", "12.34.56.78", "1.2.3.4:123" }
+      for _, name in ipairs(names) do
+        valid, errors, check = validate_entity({ target = name }, targets_schema)
+        assert.True(valid)
+        assert.is_nil(errors)
+        assert.is_nil(check)
+      end
+
+      valid, errors, check = validate_entity({ target = "\\\\bad\\\\////name////" }, targets_schema)
+      assert.False(valid)
+      assert.Nil(errors)
+      assert.equal("Invalid target; not a valid hostname or ip address", check.message)
+
+    end)
+    
+    it("should normalize 'target' field and verify default port", function()
+      local valid, errors, check
+
+      -- the utils module does the normalization, here just check whether it is being invoked.
+      local names_in = { "012.034.056.078", "01.02.03.04:123" }
+      local names_out = { "12.34.56.78:"..default_port, "1.2.3.4:123" }
+      for i, name in ipairs(names_in) do
+        local data = { target = name }
+        valid, errors, check = validate_entity(data, targets_schema)
+        assert.True(valid)
+        assert.is_nil(errors)
+        assert.is_nil(check)
```

Comment:
This is using the `is_nil` form as we should. Better stay consistent, another reason to switch to the `is_` form altogether.